ast/compile.go

```diff
@@ -3160,21 +3157,35 @@ func reorderBodyForSafety(builtins map[string]*Builtin, arity func(Ref) int, glo
 				continue
 			}
 
-			safe.Update(outputVarsForExpr(e, arity, safe))
+			ovs := outputVarsForExpr(e, arity, safe)
```

Comment:
These are the ones we've been missing in `reorderBodyForClosures`: because there, we kept feeding `globals` as third parameter, instead of the accumulated _safe_ vars, we'd never see output variables for function calls whose inputs only become safe after reordering.